File: python/6kyu/find_the_odd_int.py
# https://www.codewars.com/kata/54da5a58ea159efa38000836
import operator
from functools import reduce


def find_it(seq):
    return reduce(operator.xor, seq)


print(find_it([20, 1, -1, 2, -2, 3, 3, 5, 5, 1, 2, 4, 20, 4, -1, -2, 5]), 5)

File: go/6kyu/find_the_odd_int.go
// https://www.codewars.com/kata/54da5a58ea159efa38000836

package main

import (
	"fmt"
)

func FindOdd(seq []int) (odd int) {
	for _, n := range seq {
		odd ^= n
	}

	return
}

func main() {
	fmt.Println(FindOdd([]int{20, 1, -1, 2, -2, 3, 3, 5, 5, 1, 2, 4, 20, 4, -1, -2, 5}), 5)
}
